substrate/client/transaction-pool/src/fork_aware_txpool/view.rs

```diff
@@ -455,4 +454,10 @@ where
 			);
 		}
 	}
+
+	/// Returns true if the transaction with given hash is already imported into the view.
+	pub(super) fn is_imported(&self, tx_hash: &ExtrinsicHash<ChainApi>) -> bool {
+		const IGNORE_BANNED: bool = false;
+		self.pool.validated_pool().check_is_known(tx_hash, IGNORE_BANNED).is_err()
```

Comment:
nit: I think the logic could be improved here from a reading standpoint: e.g. calling `is_err()` at the end to determine if the tx `is_imported` sends mixed signals. IMO `check_is_known` should be renamed to `is_imported`, which checks only this, and then add a new method for `is_temporarily_banned`, which verifies the other particularity of the `check_is_known` logic (both in `Pool` impl). Then have this wrapper function in the `View` impl which depends on `is_imported` from the `Pool` impl. WDYT?

Reply:
yeah, `check_is_known` logic is a bit twisted, but again, do not want to get into renaming rabbit hole in this PR. I hope `is_imported -> bool` is readable enough.

Maybe I'll touch it in some follow.

Reply:
A follow up  sounds good